library/src/main/java/com/datatorrent/lib/state/managed/ManagedTimeUnifiedStateImpl.java

```diff
@@ -0,0 +1,238 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.state.managed;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.RemoteIterator;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Queues;
+import com.google.common.util.concurrent.Futures;
+
+import com.datatorrent.lib.fileaccess.FileAccess;
+import com.datatorrent.lib.state.BucketedState;
+import com.datatorrent.netlet.util.Slice;
+
+/**
+ * In this implementation of {@link ManagedState} the buckets in memory are time-buckets.
+ * <p/>
+ *
+ * <b>Difference from {@link ManagedTimeStateImpl}</b>: <br/>
+ * <ol>
+ * <li>The main buckets in {@link ManagedTimeStateImpl} are unique adhoc long ids which the user provides with the
+ * key. In this implementation the main buckets are time buckets. The user provides just the time and the time bucket is
+ * derived from it.
+ * </li>
+ * <br/>
+ *
+ * <li>In regards to the bucket data on disk, in {@link ManagedTimeStateImpl} the buckets are persisted on disk
+ * with each bucket data further grouped into time-buckets: {base_path}/{bucketId}/{time-bucket id}. <br/>
+ * In this implementation operator id is used as bucketId (on disk) and there is just one time-bucket under a
+ * particular operator id:
+ * {base_path}/{operator id}/{time bucket id}.
+ * </li>
+ * <br/>
+ *
+ * <li>In {@link ManagedTimeStateImpl} a bucket belongs to just one partition. Multiple partitions cannot write to
+ * the same bucket. <br/>
+ * In this implementation multiple partitions can be working with the same time-bucket (since time-bucket is derived
+ * from time). This works because on the disk the time-bucket data is segregated under each operator id.
+ * </li>
+ * <br/>
+ *
+ * <li>While {@link ManagedTimeStateImpl} can support dynamic partitioning by pre-allocating buckets this will not
+ * be able to support dynamic partitioning efficiently.
```

Comment:
I think having the table in javadoc is better since it will be visible on our API reference page.
Here is one example (though not a real fancy table):
https://docs.oracle.com/javase/7/docs/jre/api/net/httpserver/spec/com/sun/net/httpserver/HttpServer.html

You can put it in one implementation class and just have links in the other 2.